editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -504,6 +504,80 @@ void SpriteFramesEditor::_update_show_settings() {
 	}
 }
 
+void SpriteFramesEditor::_auto_slice_sprite_sheet() {
+	if (updating_split_settings) {
+		return;
+	}
+	updating_split_settings = true;
+
+	const Size2i size = split_sheet_preview->get_texture()->get_size();
+
+	const Size2i split_sheet = _estimate_sprite_sheet_size(split_sheet_preview->get_texture());
+	split_sheet_h->set_value(split_sheet.x);
+	split_sheet_v->set_value(split_sheet.y);
+	split_sheet_size_x->set_value(size.x / split_sheet.x);
+	split_sheet_size_y->set_value(size.y / split_sheet.y);
+	split_sheet_sep_x->set_value(0);
+	split_sheet_sep_y->set_value(0);
+	split_sheet_offset_x->set_value(0);
+	split_sheet_offset_y->set_value(0);
+
+	updating_split_settings = false;
+
+	frames_selected.clear();
+	selected_count = 0;
+	last_frame_selected = -1;
+	split_sheet_preview->queue_redraw();
+}
+
+bool SpriteFramesEditor::_matches_background_color(const Color &background_color, const Color &pixel_color) {
+	if ((background_color.a == 0 && pixel_color.a == 0) || background_color.is_equal_approx(pixel_color)) {
+		return true;
+	}
+
+	return pow(background_color.r - pixel_color.r, 2) + pow(background_color.g - pixel_color.g, 2) + pow(background_color.b - pixel_color.b, 2) + pow(background_color.a - pixel_color.a, 2) < 0.04;
+}
+
+Size2i SpriteFramesEditor::_estimate_sprite_sheet_size(const Ref<Texture2D> texture) {
+	Ref<Image> image = texture->get_image();
+	Size2i size = texture->get_size();
+
+	Color assumed_background_color = image->get_pixel(0, 0);
+	Size2i sheet_size = Size2i(0, 0);
```

Comment:
```suggestion
	Size2i sheet_size;
```
Unnecessary default

Reply:
Will `Size2i` default to 0,0 even if it's not initialised? Because it really does need to start at 0, 0 in this code so it can be incremented.

Reply:
Done, in next push.